### ISSUE TITLE
Saving back to ifc file will fail when ifcGrid or ifcGridAxis are removed

### ISSUE THREAD

**Issue by lbj** [IfcOpenShell/IfcOpenShell#1925]
BlenderBIM version (blenderbim-211203)
Blender version 3.0

To reproduce: Open an ifc file (this was 2x3) with several ifcGrids located under building storeys. Delete one of them in the outliner. (In this case an ifcGrid instance, but it will fail on ifcGridAxis as well) Then try to save back to ifc. 

Error message: 
Python: Traceback (most recent call last):
  File "/home/usr/.config/blender/3.0/scripts/addons/blenderbim/bim/module/project/operator.py", line 754, in invoke
    return self.execute(context)
  File "/home/usr/.config/blender/3.0/scripts/addons/blenderbim/bim/module/project/operator.py", line 762, in execute
    return IfcStore.execute_ifc_operator(self, context)
  File "/home/usr/.config/blender/3.0/scripts/addons/blenderbim/bim/ifc.py", line 278, in execute_ifc_operator
    result = getattr(operator, "_execute")(context)
  File "/home/usr/.config/blender/3.0/scripts/addons/blenderbim/bim/module/project/operator.py", line 789, in _execute
    ifc_exporter.export()
  File "/home/usr/.config/blender/3.0/scripts/addons/blenderbim/bim/export_ifc.py", line 48, in export
    self.sync_all_objects()
  File "/home/usr/.config/blender/3.0/scripts/addons/blenderbim/bim/export_ifc.py", line 109, in sync_all_objects
    tool.Collector.sync(obj)
  File "/home/usr/.config/blender/3.0/scripts/addons/blenderbim/tool/collector.py", line 46, in sync
    if obj.users_collection != (object_collection,) and obj.users_collection[0].name != obj.name:
IndexError: tuple index out of range

**Comment by Moult** [comment]
@lbj I suspect this is not actually to do with deleting grids, but instead suggests there is a tracked object that is not in the blender collection system. This could either be a normal phenomenon which I underestimated, or it could be a symptom of a deeper problem.

Can you send me your file? If it is private dion@thinkmoult.com can work. I cannot seem to replicate the issue here on a fresh file.

**Comment by lbj** [comment]
@Moult It might be related to incorrect use of BlenderBIM but the only change I've done since I opened the file was to delete the following node from the outliner: 
![image](https://user-images.githubusercontent.com/1142910/145546131-b9931f84-e378-4872-878b-7b1dbf83176f.png)
I tested saving the file after opening, and before changing anything, and I got no errors. I'll send you the file, so you can try to reproduce.

**Comment by Moult** [comment]
Happy holidays and sorry for the delay before looking at this @lbj.

Thanks for your file! I was able to replicate the issue and fix it.

Unfortunately right now the bot builds are out of sync, so if you are looking for a solution before the next build is ready, you can use this workaround:

 1. Load IFC
 2. Filter "ifcgrid" in the outliner filter input dialog
 3. Right click on an IfcGrid collection (like you have selected in blue in your screenshot) and choose Select Objects
 4. Press X to delete (I tested both with the mouse over the outliner, as well as the mouse over the 3D viewport, which are theoretically two different types of deletes, both should work)

Basically the only difference is that this explicitly select all subobjects.
